Fix geo uri google earth 11795

Rawdyrathaur opened the pull request:
## Changes
- Custom app chooser dialog to detect which app user selects
- Google Earth receives geo URIs without location names (coordinates only)
- Other apps like Google Maps receive full URIs with location names
- Optional `includeName` parameter in C++ `GenerateGeoUri()` with backward compatibility
- New JNI method `nativeGetGeoUriForApp()` for app-specific URI generation
- C++ unit tests for both URI generation modes

## Testing
- [x] Build successful
- [x] Tested on emulator with custom chooser dialog
- [x] Unit tests pass
- [x] Manual test with Google Earth
- [x] Manual test with Google Maps

## Test Environment
- Android API Level: 36
- Device: Medium Phone Emulator
- Build variant: fdroid-debug

#After


https://github.com/user-attachments/assets/c712a973-88b9-45aa-a716-82a950308163

Rawdyrathaur commented:
I wrote the code myself. However, I should have researched the problem more thoroughly before submitting. I apologize for the premature PR.
I will revert them immediately. I apologize for including them.

AndrewShkrob reviewed (changes requested):
1. Would you, as a user of the app, be satisfied with the UI of this alert dialog? I personally wouldn’t.
2. I think the only possible fix is to remove "(label)" from intent string. The only app (at least I'm aware of) where it is used is Google Maps. The label is not even a part of Android's geo intent scheme, it is only gmaps' extension. And they can work without it, however they will display "Unknown Place" most of the time. But other apps would be able to display the content (I believe not only Google Earth may be affected). 
3. Providing different intents for different apps is a dead-end. There is an official scheme that must be supported by all apps. 
4. I'd recommend reporting a bug to Google Earth app because there is literally no sense in fixing it on our side

Rawdyrathaur commented:
Thank you for the feedback. I apologize for overcomplicating the initial approach.
I've simplified it based on your suggestions - removed all the app-specific logic and just removed the label parameter from geo URIs as recommended. Much cleaner now.

Tested on emulator and works correctly.

biodranik commented:
If I understood @AndrewShkrob's idea, we need to google how to pass custom data (bookmark name in our case) to Google Maps in the intent. And then roll back the change that added () and pass that custom data so Google Maps will show a pin with a name. And Google Earth will still work with coordinates. If a name can be passed to Google Earth too in the same or similar way, that would be even better.
